api/src/main/java/org/openmrs/module/referencedemodata/ReferenceDemoDataActivator.java

```diff
@@ -135,7 +135,7 @@ public void started() {
 						boolean isInProgram = false;
 						
 						Patient patient = Context.getPatientService().getPatient(patientId);
-						int visitCount = randomBetween(1, Math.min(Math.round(patient.getAge() / 1.5f), 10));
+						int visitCount = randomBetween(1, Math.max(Math.round(patient.getAge() / 10.5f), 1));
```

Comment:
Any reason to generate substantially fewer visits?

Reply:
This may rather not mean fewer visits but, instead of having 10 maximum visits for all patients above 15 years, let the number be fairly incremented based on the patient's age. That is the older the patient, the closer to 10 number of visits they can have(This is because there  is no patient with age greater than 100). Otherwise `randomBetween ()` tends to generate more maximum values in the range - and having 10 visits on 90% of generated patients exponentially increase overall demo-data generation.

Reply:
Ok...